### DISCUSSION TITLE
Error - StyleAdapter and cfg/guess mode may not works due to non-batch-cond inference (clarification)

### DISCUSSION THREAD

**Discussion by tnginako** [Mikubill/sd-webui-controlnet#521] [Q&A] [answered]
Hello, just wanna ask if anyone else is getting this error? "StyleAdapter and cfg/guess mode may not works due to non-batch-cond inference". I can use Controlnet fine but this might be affecting the results or something. I tried searching this error on past discussion but doesn't seem to appear.
![image](https://user-images.githubusercontent.com/123961293/223399306-3d3c5156-a502-425c-b56a-07e030a42260.png)

Edit: Would this be related to this setting? I think I haven't enabled it yet.
![image](https://user-images.githubusercontent.com/123961293/223401542-87e62e66-c4be-4e81-9231-d4d458cc841c.png)

**Comment by slashedstar** [comment, marked as answer]
Yeah, the warning shows as soon as you exceed 75 tokens in the prompt (and it doesn't apply the style)

**Reply by tnginako** [reply]
Ohhh thanks! Yeah my prompt was long and token was more than 75 when the error appeared.

**Reply by AIAMIAUTHOR** [reply]
weird, i always use more than 75 tokens, i have been using --always-batch-cond-uncond, guess mode, enable optimized monocular depth estimation, enable CFG-Based guidance, cfg 5, steps 50, without this message or problems, this error is new for me:  StyleAdapter and cfg/guess mode may not works due to non-batch-cond inference, i started getting this error just some hours ago too, probably related with 75 tokens also breaking the StyleAdapter model, compared last commit with new one, it thinks it's kinda affecting some generations. this error pops up with almost all control models and t2i adapters, not only styleadapter, probably just a temp warning message?

**Reply by renatodigiorgio** [reply]
same issue here. i remove all prompts (using img2img with 3 controlnets activate: cany + hed + t2iadapter with clip_vision preprocessor), in generating process appears error: "warning: StyleAdapter and cfg/guess mode may not works due to non-batch-cond inference" and generated result appears with not affected style =( frustrating .... i try many denoising strengths in img2img and many weights on controlnet instances without success... not applying the style on final generated result =(

**Reply by Sludge-Demon** [reply]
Removing most of my negative prompt solved it for me.

**Comment by f-rank** [comment]
I'm getting this while not using Controlnet. Can't the prompts go over 75 if no Controlnet is on ?
CFG based option is off.

**Comment by soundtrackgeek** [comment]
Have the same issue. My prompts are 74/75, Tried turning on CFG in Controlnet settings, but still the same problem.

**Comment by rafaelfabres** [comment]
Same issue!

**Comment by SeBL4RD** [comment]
Same issue, 
following a comment on Youtube, I can confirm that with `--medvram` enabled, even with less than 75 tokens, the error message appears and controlNet does not work.
_Warning: StyleAdapter and cfg/guess mode may not works due to non-batch-cond inference_

when I deactivate `--medvram`, everything works flawlessly again, i can use _t2iadapter_style_, but I can't exceed 768x768 pixels... I only have 8 GB of Vram...

**Reply by renatodigiorgio** [reply]
i will try... i have --medvram activated... only 6gb of gpu here =( rtx2060

**Reply by renatodigiorgio** [reply]
you gotcha @SeBL4RD !!! works now without --medvram ... thanks _/\_ =)

**Reply by soundtrackgeek** [reply]
Oh interesting. I have a 3060 6gb vram, so --medvram is pretty much essential, but I'll give it a go. Thanks!

**Reply by AIAMIAUTHOR** [reply]
functions (--opt-split-attention --always-batch-cond-uncond --medvram) are essential for some users and they should be supported, also some generations w/o xformers created with 10xx, 20xx,30xxx drivers and hardware are unique.

**Reply by Waykasa** [reply]
There was a problem that the clip vision with the adapter style refused to work. Deactivating `--medvram` helped. Thanks a lot!

**Reply by amoghvaishampayan** [reply]
How do I deactivate medvram through the Automatic1111 settings? I am running it on Colab.

**Comment by MrDo1965** [comment]
For me adding --always-batch-cond-uncond do the trick as AIAMIAUTHOR suggest
now webui-user.bat start with:
set COMMANDLINE_ARGS= --lowvram --xformers --always-batch-cond-uncond
and also "Enable CFG-Based guidance" in settings was ticked on. Don't know is this nedeed too.
So this error "Error - StyleAdapter and cfg/guess mode may not works due to non-batch-cond inference" goes away and style was applied.

**Reply by SeBL4RD** [reply]
Really???
Are you sure `--always-batch-cond-uncond` does not have a harmful effect? what is it supposed to be used for?

**Reply by SeBL4RD** [reply]
Its WORKING, YEAH !!!

**Reply by rafaelfabres** [reply]
But do you still need to use less than 75 tokens?

**Reply by SeBL4RD** [reply]
> But do you still need to use less than 75 tokens?

Yes, you have to.

**Reply by soundtrackgeek** [reply]
Finally! Actually in my tests, I had 90 tokens for prompt and 134 for negative prompt. Worked fine for me.

**Reply by hoodady** [reply]
can confirm worked for me too, thanks guys

**Reply by Myoko** [reply]
not work for me! i had 34 prompt and 234 negative prompt

**Reply by AIAMIAUTHOR** [reply]
> not work for me! i had 34 prompt and 234 negative prompt

StyleAdapter and cfg/guess mode may not works due to non-batch-cond inference is still a problem for a lot of users, still not sure if its just a warning or a real problem discarding prompt tokens, with or without commandline arg the problems appears.

**Reply by drmmkyr** [reply]
i also facing the same issue, i using DPM++ SDE Karras for sampling, if the positive and negative token not the same level(75/150), the error will occurred and Controlnet will not apply to the result. so after i adjust both prompt to same level the problem solve.

**Comment by bassmanjazz** [comment]
Keeping both postive and negative prompts under 75 tokens didn't give me the error.

**Comment by renatodigiorgio** [comment]
you have to edit the startup file after run it removing this
but i think in colab not necessary do this

Em sáb., 15 de abr. de 2023 às 10:22, Amogh Vaishampayan <
***@***.***> escreveu:

> How do I deactivate medvram through the Automatic1111 settings? I am
> running it on Colab.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Mikubill/sd-webui-controlnet/discussions/521#discussioncomment-5623536>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG33I24SMSNZVWQJ3YWOWKDXBKOJTANCNFSM6AAAAAAVSJPDVI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> .com>
>


-- 
*Renato Di Giorgio*
*http://renatodigiorgio.com <http://renatodigiorgio.com>*
***@***.*** ***@***.***>*
11 989636255
11 30426677
